Samples/MobileBuyIntegration/MobileBuyIntegration/ProductView.swift

```diff
@@ -0,0 +1,290 @@
+    }
+
+    // MARK: Body
+    var body: some View {
+        ScrollView {
+            VStack(spacing: 16) {
+                 if let imageURL = product.featuredImage?.url {
+					ZStack {
+						Rectangle()
+							.fill(Color.gray.opacity(0.2))
+							.frame(height: 400)
+
+						AsyncImage(url: imageURL) { phase in
+							switch phase {
+							case .empty:
+								EmptyView()
+							case .success(let image):
+								image
+									.resizable()
+									.aspectRatio(contentMode: .fill)
+									.frame(height: 400)
+									.clipped()
+									.opacity(imageLoaded ? 1 : 0)
+									.onAppear {
+										withAnimation(.easeIn(duration: 0.5)) {
+											imageLoaded = true
+										}
+									}
+							case .failure:
+								Image(systemName: "photo")
+									.resizable()
+									.frame(width: 100, height: 100)
+									.foregroundColor(.gray)
+							@unknown default:
+								EmptyView()
+							}
+						}
+					}
+					.frame(height: 400)
+				}
+
+                VStack(alignment: .leading, spacing: 8) {
+
+                    Text(product.vendor)
+                        .font(.body)
+                        .fontWeight(.semibold)
+						.padding(.vertical)
+                        .foregroundColor(Color(ColorPalette.primaryColor))
+
+					Text(product.title)
+                        .font(.title)
+
+                    Text(product.description)
+                        .font(.body)
+                        .foregroundColor(.gray)
+                        .lineLimit(descriptionExpanded ? 10 : 3)
+                        .onTapGesture {
+							descriptionExpanded.toggle()
+						}
+                }
+                .frame(maxWidth: .infinity, alignment: .leading)
+                .padding(.horizontal)
+
+                if let variant = product.variants.nodes.first {
+                    Button(action: addToCart) {
+                        HStack {
+                            Text(loading ? "Adding..." : (addedToCart ? "Added" : "Add to Cart"))
+								.font(.headline)
+
+                            if loading {
+                                ProgressView()
+									.colorInvert()
+                            }
+                            Spacer()
+
+							Text((variant.availableForSale ? (addedToCart ? "✓" : ( variant.price.formattedString())) : "Out of stock")!)
+                        }.padding()
+                    }
+					.background(addedToCart ? Color(ColorPalette.successColor) : Color(ColorPalette.primaryColor))
+					.foregroundStyle(.white)
+					.cornerRadius(10)
+                    .disabled(!variant.availableForSale || loading)
+                    .padding([.leading, .trailing], 15)
+                }
+            }
+        }
+		.navigationTitle(product.collections.nodes.first?.title ?? product.title)
+    }
+
+    // MARK: Methods
+    private func addToCart() {
+        guard let variant = product.variants.nodes.first else { return }
+
+        loading = true
+        let start = Date()
```

Comment:
Continuous clock looks interesting.. Although I'm not sure how easy it is to use with callbacks.

```
let clock = ContinuousClock()
let elapsed = clock.measure {
    someWork()
}
```